homeassistant/components/zwave_me/__init__.py

```diff
@@ -0,0 +1,122 @@
+"""The Z-Wave-Me WS integration."""
+import logging
+
+from zwave_me_ws import ZWaveMe, ZWaveMeData
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.dispatcher import async_dispatcher_connect, dispatcher_send
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN, PLATFORMS, ZWAVEPLATFORMS
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry):
+    """Set up Z-Wave-Me from a config entry."""
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+    hass.data[DOMAIN] = ZWaveMeController(hass, entry)
+
+    return True
+
+
+async def async_unload_entry(hass, entry):
+    """Unload a config entry."""
+    return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+
+class ZWaveMeController:
+    """Main ZWave-Me API class."""
+
+    def __init__(self, hass: HomeAssistant, config: ConfigEntry) -> None:
+        """Create the API instance."""
+        self.device_ids: list[str] = []
+        self._hass = hass
+        self._config = config
+        self.zwave_api = ZWaveMe(
+            on_device_create=self.on_device_create,
+            on_device_update=self.on_device_update,
+            on_new_device=self.add_device,
+            token=self._config.data["token"],
+            url=self._config.data["url"],
+            platforms=ZWAVEPLATFORMS,
+        )
+
+    def add_device(self, device: ZWaveMeData):
+        """Send signal to create device."""
+        if device.deviceType in ZWAVEPLATFORMS:
+            if device.id in self.device_ids:
+                dispatcher_send(self._hass, "ZWAVE_ME_INFO_" + device.id, device)
+            else:
+                dispatcher_send(
+                    self._hass, "ZWAVE_ME_NEW_" + device.deviceType.upper(), device
+                )
+                self.device_ids.append(device.id)
+
+    def on_device_create(self, devices: list[ZWaveMeData]):
+        """Create multiple devices."""
+        if devices is None:
+            return
+        for device in devices:
+            self.add_device(device)
+
+    def on_device_update(self, new_info: ZWaveMeData):
+        """Send signal to update device."""
+        dispatcher_send(self._hass, "ZWAVE_ME_INFO_" + new_info.id, new_info)
+
+
+class ZWaveMeDevice(Entity):
+    """Representation of a ZWaveMe device."""
+
+    def __init__(self, device):
+        """Initialize the device."""
+        self._name = device.title
+        self._outlet = None
+        self.device = device
+
+    async def async_added_to_hass(self) -> None:
+        """Connect to an updater."""
+        async_dispatcher_connect(
```

Comment:
Unsubscribe this dispatcher when the entity is removed.
```suggestion
        self.async_on_remove(async_dispatcher_connect(
```

(and add closing `)` after this call)